test/src/api/responder03Test.php

```diff
@@ -4173,4 +4173,303 @@ public function test_interface_game_058() {
+        $expData['playerDataArray'][0]['activeDieArray'][0]['sides'] = 4;
+        $expData['playerDataArray'][0]['activeDieArray'][0]['skills'] = array("Mood");
+        $expData['playerDataArray'][0]['activeDieArray'][0]['value'] = 3;
+        $expData['playerDataArray'][0]['activeDieArray'][1]['description'] = "Doppelganger 12-sided die";
+        $expData['playerDataArray'][0]['activeDieArray'][1]['properties'] = array();
+        $expData['playerDataArray'][0]['activeDieArray'][1]['recipe'] = "D(12)";
+        $expData['playerDataArray'][0]['activeDieArray'][1]['sides'] = 12;
+        $expData['playerDataArray'][0]['activeDieArray'][1]['skills'] = array("Doppelganger");
+        $expData['playerDataArray'][0]['activeDieArray'][1]['value'] = 10;
+        $expData['playerDataArray'][0]['activeDieArray'][2]['description'] = "Doppelganger X Swing Die (with 15 sides)";
+        $expData['playerDataArray'][0]['activeDieArray'][2]['recipe'] = "D(X)";
+        $expData['playerDataArray'][0]['activeDieArray'][2]['sides'] = 15;
+        $expData['playerDataArray'][0]['activeDieArray'][2]['value'] = 15;
+        array_pop($expData['playerDataArray'][0]['activeDieArray']);
+        $expData['playerDataArray'][0]['capturedDieArray'][0]['properties'] = array();
+        $expData['playerDataArray'][0]['roundScore'] = 25.5;
+        $expData['playerDataArray'][0]['sideScore'] = 1.7;
+        $expData['playerDataArray'][0]['waitingOnAction'] = true;
+        $expData['playerDataArray'][1]['activeDieArray'][1]['description'] = "Morphing 6-sided die";
+        $expData['playerDataArray'][1]['activeDieArray'][1]['properties'] = array("HasJustMorphed");
+        $expData['playerDataArray'][1]['activeDieArray'][1]['recipe'] = "m(6)";
+        $expData['playerDataArray'][1]['activeDieArray'][1]['sides'] = 6;
+        $expData['playerDataArray'][1]['activeDieArray'][1]['value'] = 6;
+        $expData['playerDataArray'][1]['activeDieArray'][2]['value'] = 1;
+        $expData['playerDataArray'][1]['capturedDieArray'][1]['properties'] = array("WasJustCaptured");
+        $expData['playerDataArray'][1]['capturedDieArray'][1]['recipe'] = "D(6)";
+        $expData['playerDataArray'][1]['capturedDieArray'][1]['sides'] = 6;
+        $expData['playerDataArray'][1]['capturedDieArray'][1]['value'] = 6;
+        $expData['playerDataArray'][1]['roundScore'] = 23;
+        $expData['playerDataArray'][1]['sideScore'] = -1.7;
+        $expData['playerDataArray'][1]['waitingOnAction'] = false;
+        $expData['validAttackTypeArray'] = array("Power");
+
+        $retval = $this->verify_api_loadGameData($expData, $gameId, 10);
+
+        $this->verify_api_submitTurn(
+            array(9),
+            'responder003 performed Power attack using [D(X=15):15] against [(X=10)!:4]; Defender (X=10)! was captured; Attacker D(X=15) changed size from 15 to 10 sides, recipe changed from D(X=15) to (X=10)!, rerolled from 15. ',
```

Comment:
So is this the behavior we want, where it says what the die is rerolling from, but not what it's rerolling to?

Reply:
My understanding is that there *should* be a follow-up action log with the value it rerolls to.

The reason for the split is that the two happen in different game phases, with the value assigned after the turbo size has been set/determined.

Reply:
Assuming I'm reading the responder test correctly, it seems that the follow-up log entry isn't there. That's not ideal. I'll take a look and see why the log doesn't contain the follow-up message.